robopi:
  diff_drive_controller:
     type: "diff_drive_controller/DiffDriveController"
     publish_rate: 50
     publish_cmd: true
     enable_odom_tf: true

     left_wheel: ['left__wheel_joint']
     right_wheel: ['right__wheel_joint']

     # Odometry covariances for the encoder output of the robot. These values should
     # be tuned to your robot's sample odometry data, but these values are a good place
     # to start
     pose_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
     twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]

     # Top level frame (link) of the robot description
     base_frame_id: base_link
 
     # Velocity and acceleration limits for the robot
     linear:
       x:
         has_velocity_limits    : true
         max_velocity           : 0.1  # m/s
         # min_velocity           : defaults to -*max velocity
         has_acceleration_limits: true
         max_acceleration       : 0.5   # m/s^2
     angular:
       z:
         has_velocity_limits    : true
         max_velocity           : 10   # rad/s
         # min_velocity           : defaults to -*max velocity
         has_acceleration_limits: true
         max_acceleration       : 0.5   # rad/s^2

  joint_state_controller:
    type: "joint_state_controller/JointStateController"
    publish_rate: 50

  hardware_interface_arduino:
    loop_hz: 50 # hz
    device_name : "/dev/ttyACM0"
    wheels:
      - left__wheel_joint
      - right__wheel_joint

